mesonbuild/modules/python.py

```diff
@@ -332,7 +332,9 @@ def _find_installation_impl(self, state: 'ModuleState', display_name: str, name_
             # named python is available and has a compatible version, let's use
             # it
             if not python.found() and name_or_path in {'python2', 'python3'}:
-                python = PythonExternalProgram('python')
+                python_cmd = shutil.which('python')
+                if python_cmd:
+                    python = PythonExternalProgram(name_or_path, command=python_cmd)
```

Comment:
Hmm, thinking more about this... this is the same problem we solve with `tmp_python` above.

But there's actually one distinction between ExternalProgram and shutil.which -- we ignore WindowsApps reparse points that might just open the Windows store, see commit 64432e94481dbd55bedf92f21610bc5ed6b52655.

I think we can't get around using ExternalProgram here instead of shutil.which. It's still advantageous to avoid duplicating the code around python.sanity() and checking required and so on.

Reply:
Did the change. 

Still not sure if the error message is clear enough when it finds a python that is python3 when looking for python2.

Reply:
Then other code doesn't bother with tmp_python.found() because it will just reuse the tmp_python's found status. So this can probably be a bit shorter although overall doesn't matter to the end result.

Reply:
I still have a `Program nonexistingprogram found: NO` displayed. I will try to track it

Reply:
I added a commit to fix #11686 .

Reply:
After that change, the `required: true` error path has reduced logging.

Reply:
Also please move "Fixes" out of the one-line summary and into the body. :)

Reply:
Well. Having both the log and the exception is maybe not so bad after all...

> Also please move "Fixes" out of the one-line summary and into the body. :)

Still learning... ;-)